src/main/java/com/myapp/impl/ItemDaoImpl.java

```diff
@@ -0,0 +1,107 @@
+package com.myapp.impl;
+
+import com.myapp.dao.ItemDao;
+import com.myapp.model.Item;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementCreator;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Component;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+@Component
```

Comment:
https://www.baeldung.com/spring-component-repository-service

better to choose the annotation-based on their layer conventions
here better suited annotation is @repository